crates/signature-mpc/src/twopc_mpc_protocols/encrypt_user_share.rs

```diff
@@ -0,0 +1,406 @@
+        Err(e) => {
+            return Err(e);
+        }
+    };
+
+    Ok(EncryptedUserShareAndProof {
+        proof: proofs,
+        encrypted_user_share: statements[0]
+            .language_statement()
+            .encrypted_discrete_log()
+            .value(),
+        range_proof_commitment: statements[0].range_proof_commitment().value(),
+    })
+}
+
+fn generate_witnesses(
+    randomness: RandomnessSpaceGroupElement,
+    enhanced_language_public_parameters: &language::PublicParameters<
+        { maurer::SOUND_PROOFS_REPETITIONS },
+        EnhancedLang<
+            { maurer::SOUND_PROOFS_REPETITIONS },
+            RANGE_CLAIMS_PER_SCALAR,
+            RandomnessSpaceGroupElement,
+            Lang,
+        >,
+    >,
+    plaintext: PlaintextSpaceGroupElement,
+) -> Vec<
+    WitnessSpaceGroupElement<
+        { maurer::SOUND_PROOFS_REPETITIONS },
+        RANGE_CLAIMS_PER_SCALAR,
+        COMMITMENT_SCHEME_MESSAGE_SPACE_SCALAR_LIMBS,
+        bulletproofs::RangeProof,
+        RandomnessSpaceGroupElement,
+        Lang,
+    >,
+> {
+    let witnesses: Vec<language::WitnessSpaceGroupElement<1, Lang>> =
+        vec![(plaintext, randomness).into()];
+
+    let witnesses =
+        EnhancedLanguage::<
+            { maurer::SOUND_PROOFS_REPETITIONS },
+            RANGE_CLAIMS_PER_SCALAR,
+            { COMMITMENT_SCHEME_MESSAGE_SPACE_SCALAR_LIMBS },
+            bulletproofs::RangeProof,
+            RandomnessSpaceGroupElement,
+            Lang,
+        >::generate_witnesses(witnesses, &enhanced_language_public_parameters, &mut OsRng)
+        .unwrap();
+    witnesses
+}
+
+pub fn enhanced_language_public_parameters<
+    const REPETITIONS: usize,
+    const NUM_RANGE_CLAIMS: usize,
+    UnboundedWitnessSpaceGroupElement: group::GroupElement + Samplable,
+    Lang: EnhanceableLanguage<
+        REPETITIONS,
+        NUM_RANGE_CLAIMS,
+        COMMITMENT_SCHEME_MESSAGE_SPACE_SCALAR_LIMBS,
+        UnboundedWitnessSpaceGroupElement,
+    >,
+>(
+    unbounded_witness_public_parameters: UnboundedWitnessSpaceGroupElement::PublicParameters,
+    language_public_parameters: Lang::PublicParameters,
+) -> language::PublicParameters<
+    REPETITIONS,
+    EnhancedLang<REPETITIONS, NUM_RANGE_CLAIMS, UnboundedWitnessSpaceGroupElement, Lang>,
+> {
+    MaurerPublicParameters::new::<
+        bulletproofs::RangeProof,
+        UnboundedWitnessSpaceGroupElement,
+        Lang,
+    >(
+        unbounded_witness_public_parameters,
+        bulletproofs::PublicParameters::default(),
+        language_public_parameters,
+    )
+    .unwrap()
+}
+pub fn is_valid_proof(
+    language_public_parameters: LangPublicParams,
+    proof_public_output: EncryptedUserShareAndProof,
+    centralized_public_keyshare: group::Value<secp256k1::GroupElement>,
+) -> bool {
+    let secp256k1_group_public_parameters = secp256k1::group_element::PublicParameters::default();
+    let protocol_public_parameters = ProtocolPublicParameters::new(N);
```

Comment:
Don't use test constants in production code: https://github.com/dwallet-labs/dwallet-network/pull/123/files/e3e7e8963c1181ad84bb8e3b42288f6816e34468#r1705313423